.github/workflows/production-build.yaml

```diff
@@ -15,11 +21,18 @@ jobs:
         service_name: [backend, frontend, document-service, platform-service, prompt-service, worker, x2text-service]
 
     steps:
-      - name: Checkout code
+      - name: Checkout code for release
+        if: github.event_name == 'release'
         uses: actions/checkout@v2
         with:
           ref: ${{ github.event.release.tag_name }}
 
+
+      - name: Checkout code for branch
+        if: github.event_name != 'release'
```

Comment:
@johnyrahul reference should be the branch.

Reply:
branch will get reffered while running the workflow manually.  Is my understanding correct
?

Reply:
Addressed

Reply:
@jaseemjaskp no need. Branch by default is selected as part of workflow run. So the selected branch will be checked out.